modules/server/src/network/resolvers/aggregations.ts

```diff
@@ -1,101 +1,120 @@
+		const response = await queryConnection(query);
+		return { config, response };
+	});
 
-/**
- * Creates resolver function
- * @param field
- * @returns resolver function
- */
-const createResolver = (field: SupportedNetworkFieldType, configs: NetworkAggregationConfig[]) => {
-	const connectionsToQuery = getConnectionURLs(field, configs);
-
-	return async () => {
-		const connectionResponses = await Promise.allSettled(
-			connectionsToQuery.map((url: string) => queryRemoteConnection(url, field.type)),
-		);
-
-		const responses = connectionResponses.filter(
-			(result: { status: string }) => result.status === 'fulfilled',
-		);
+	// TODO: expand on network condition handling, eg. timeouts, single connection failure
+	const networkResults = await Promise.allSettled(networkQueryPromises);
+	return networkResults;
+};
 
-		const resolvedData = resolveAggregation(responses, field.type);
+type NetworkQuery = {
+	url: string;
+	gqlQuery: DocumentNode;
+};
 
-		/**
-		 * TODO: return resolvedData
-		 */
-		return { test: field.name };
-	};
+/**
+ * Find requested field in remote connection supported fields
+ *
+ * @param config
+ * @param fieldName
+ * @returns
+ */
+const findMatchedAggregationField = (config: NetworkAggregationConfig, fieldName: string) => {
+	return config.supportedAggregations.find((agg) => agg.name === fieldName);
 };
 
 /**
- * Returns a resolver map of functions
+ * Parse central query and build individual queries for remote connections based on available fields
+ *
  * @param configs
- * @param allTypeDefs
- * @returns resolver map
+ * @param info
+ * @returns
  */
-export const createAggregationResolvers = (
+export const createNetworkQueries = (
 	configs: NetworkAggregationConfig[],
-	networkFieldTypes: SupportedNetworkFieldType[],
-) => {
-	return networkFieldTypes.reduce(
-		(resolvers, field) => ({ ...resolvers, [field.name]: createResolver(field, configs) }),
-		{},
-	);
+	info: GraphQLResolveInfo,
+): NetworkQuery[] => {
+	const rootQueryFields = getRootFields(info);
+
+	const queries = configs
+		.map((config) => {
+			// construct gql string { [fieldName] { [Aggregation query fields] } }
+			const gqlString = rootQueryFields.reduce((gqlString, fieldName) => {
+				const matchedAggregationField = findMatchedAggregationField(config, fieldName);
+				if (matchedAggregationField) {
+					const { name, type } = matchedAggregationField;
+					// get gql query string for supported aggregation
+					const aggregationFieldQueryString = supportedAggregationQueries.get(type);
+					return gqlString + `${name}${aggregationFieldQueryString}`;
+				}
+				return gqlString;
+			}, '');
+
+			/*
+			 * convert string to AST object to use as query
+			 * not needed if gqlString is formatted correctly but this acts as a validity check
+			 */
+			try {
+				const gqlQuery = gql`{${gqlString}}`;
+				return { url: config.graphqlUrl, gqlQuery };
+			} catch (err) {
+				console.error(err);
+				return false;
+			}
```

Comment:
I'm not sure what to suggest here, but it seems weird that a whole remote node might just get no query sent because the gql threw an error. Shouldn't we have an object that indicates there is no gql query for that node and why?